source/extensions/filters/network/common/redis/redis_command_stats.cc

```diff
@@ -0,0 +1,106 @@
+#include "extensions/filters/network/common/redis/redis_command_stats.h"
+
+#include "extensions/filters/network/common/redis/supported_commands.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace Common {
+namespace Redis {
+
+RedisCommandStats::RedisCommandStats(Stats::Scope& scope, const std::string& prefix, bool enabled)
+    : scope_(scope), stat_name_set_(scope.symbolTable()), prefix_(stat_name_set_.add(prefix)),
+      enabled_(enabled), upstream_rq_time_(stat_name_set_.getStatName(upstream_rq_time_metric_)) {
+  // Note: Even if this is disabled, we track the upstream_rq_time.
+  stat_name_set_.rememberBuiltin(upstream_rq_time_metric_);
+
+  if (enabled_) {
+    // Create StatName for each Redis command. Note that we don't include Auth or Ping.
+    for (const std::string& command :
+         Extensions::NetworkFilters::Common::Redis::SupportedCommands::simpleCommands()) {
+      createStats(command);
+    }
+    for (const std::string& command :
+         Extensions::NetworkFilters::Common::Redis::SupportedCommands::evalCommands()) {
+      createStats(command);
+    }
+    for (const std::string& command : Extensions::NetworkFilters::Common::Redis::SupportedCommands::
+             hashMultipleSumResultCommands()) {
+      createStats(command);
+    }
+    createStats(Extensions::NetworkFilters::Common::Redis::SupportedCommands::mget());
+    createStats(Extensions::NetworkFilters::Common::Redis::SupportedCommands::mset());
+  }
+}
+
+void RedisCommandStats::createStats(std::string name) {
+  stat_name_set_.rememberBuiltin(name + ".total");
```

Comment:
prefer to put
```
  const StatName total_;
  const StatName success_;
  ...
```
in your class, populating at construction time via StatNameSet::add() or StatNamePool::add(). Then you can compose them into full stat-names via join().

Reply:
Sounds good. Per my other comment, `::rememberBuiltin(std::string)` and `add(std::string)` behave similarly, especially with regard to composing via `join()`?

Reply:
They are related but one of them returns a managed but unmapped StatName and the other records a string-to-StatName map entry and does not return anything.

Please check the comments in that class for more detail.